api/src/conf.rs

```diff
@@ -0,0 +1,95 @@
+use config::{Config, File};
+
+pub struct DatabaseConfig {
+    pub name: String,
+    pub host: String,
+    pub port: u16,
+    pub username: String,
+    pub password: String,
+}
+
+impl Default for DatabaseConfig {
+    fn default() -> Self {
+        DatabaseConfig {
+            name: "boom".to_string(),
+            host: "localhost".to_string(),
+            port: 27017,
+            username: "mongoadmin".to_string(),
+            password: "mongoadminsecret".to_string(),
+        }
+    }
+}
+
+pub struct AppConfig {
+    pub database: DatabaseConfig,
+}
+
+impl AppConfig {
+    pub fn from_default_path() -> Self {
+        load_config(None)
+    }
+
+    pub fn from_path(config_path: &str) -> Self {
+        load_config(Some(config_path))
+    }
+}
+
+impl Default for AppConfig {
```

Comment:
Should I be implementing this trait here, or should I just put this in the `AppConfig` implementation?